### DISCUSSION TITLE
@blueprintjs/datetime2 in Beta?

### DISCUSSION THREAD

**Discussion by justinbhopper** [palantir/blueprint#5819] [Q&A] [answered]
`@blueprintjs/datetime2` is currently v0.9.6.  Are there known feature gaps before it reaches 1.0?  Any reason I should wait to migrate my components before it reaches 1.0+?

**Comment by adidahiya** [comment, marked as answer]
Sorry for the delay on this - @blueprintjs/datetime2 v1.0.0 was [released](https://github.com/palantir/blueprint/commit/d6e99fbc96b1ed6ad443ed385e34cd3f7c78de10) in June 2023 with the breaking changes described in [datetime2 component migration](https://github.com/palantir/blueprint/wiki/datetime2-component-migration). most importantly:
- all usage of `{ Popover } from "@blueprintjs/core@4.x"` was migrated to `{ Popover2 } from "@blueprintjs/popover2@1.x"` (same as `{ Popover } from "@blueprintjs/core@5.x"`)
- DateInput now uses ISO strings instead of `Date` object values

At the time it was released, I was unsure whether I would be able to make the same `value` type API change (`Date` -> ISO string) to DateRangeInput to enable better timezone support, but ultimately found that doing so would delay the rollout of popper.js v2.x so I opted to postpone that work.

Now I am gearing up for another major version release of @blueprintjs/datetime2 v2.0, this time to enable react-day-picker v8 & React 18 support, see https://github.com/palantir/blueprint/issues/5699. You will be able to upgrade to those new components by following [this migration guide](https://github.com/palantir/blueprint/wiki/react-day-picker-8-migration).

**Comment by ggdouglas** [comment]
Closing this out. Both v1 and v2 of `@blueprintjs/datetime2` shipped. New code should import directly from `@blueprintjs/datetime`.
